components/brave_wallet_ui/page/screens/onboarding/onboarding.routes.tsx

```diff
@@ -0,0 +1,128 @@
+import { OnboardingSuccess } from './onboarding-success/onboarding-success'
+
+export const OnboardingRoutes = () => {
+ // redux
+ const isWalletCreated = useSelector(({ wallet }: { wallet: WalletState }) => wallet.isWalletCreated)
+ const walletTermsAcknowledged = useSelector(({ page }: { page: PageState }) => page.walletTermsAcknowledged)
+
+  // render
+
+  if (!walletTermsAcknowledged) {
+    return (
+      <Switch>
+        <Route path={WalletRoutes.OnboardingWelcome} exact>
+          <OnboardingWelcome />
+        </Route>
+
+        <Route path={WalletRoutes.Onboarding} exact>
+          <Redirect to={WalletRoutes.OnboardingWelcome} />
+        </Route>
+
+        <Redirect to={WalletRoutes.Onboarding} />
+      </Switch>
+    )
+  }
+
+  return (
+    <Switch>
+
+      {isWalletCreated &&
+        <Route path={WalletRoutes.OnboardingComplete} exact>
+          <OnboardingSuccess />
+        </Route>
+      }
+
+      <Route path={WalletRoutes.OnboardingCreatePassword} exact>
+        <OnboardingCreatePassword />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingExplainRecoveryPhrase} exact>
+        <OnboardingRecoveryPhraseExplainer />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingBackupRecoveryPhrase} exact>
+        <OnboardingBackupRecoveryPhrase />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingVerifyRecoveryPhrase} exact>
+        <OnboardingVerifyRecoveryPhrase />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingImportOrRestore} exact>
+        <OnboardingImportOrRestoreWallet />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingRestoreWallet} exact>
+        <OnboardingRestoreFromRecoveryPhrase
+          key='seed' // keys are set here to prevent holding state between page changes
+          restoreFrom='seed'
+        />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingImportMetaMask} exact>
+        <OnboardingRestoreFromRecoveryPhrase
+          key='metamask'
+          restoreFrom='metamask'
+        />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingImportMetaMaskSeed} exact>
+        <OnboardingRestoreFromRecoveryPhrase
+          key='metamask-seed'
+          restoreFrom='metamask-seed'
+        />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingImportCryptoWallets} exact>
+        <OnboardingRestoreFromRecoveryPhrase
+          key='legacy'
+          restoreFrom='legacy'
+        />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingImportCryptoWalletsSeed} exact>
+        <OnboardingRestoreFromRecoveryPhrase
+          key='legacy-seed'
+          restoreFrom='legacy-seed'
+        />
+      </Route>
+
+      <Route path={WalletRoutes.OnboardingWelcome} exact>
+        <OnboardingWelcome />
+      </Route>
+
+      <Route path={WalletRoutes.Onboarding} exact>
+        <Redirect to={WalletRoutes.OnboardingWelcome} />
+      </Route>
+
+      <Redirect to={WalletRoutes.Onboarding} />
+
+    </Switch>
+  )
```

Comment:
Now requiring that wallet terms-of-use are agreed to before certain pages are accessible